transport.rst

```diff
@@ -81,82 +81,98 @@ Example:
+The message sending node needs to know the recipient node's current ``userId`` (and therefore implicitly the node's current homeserver),
+either from a direct request to the PFS, or from address-metadata provided from a previous node (see :ref:`Discovery & Presence <transport-discovery-presence>`).
 
+Nodes are expected to  set their ``deviceId`` to ``RAIDEN``,
+so that clients sending ``toDevice`` messages have to specify ``RAIDEN`` as the target ``deviceId``.
 
-Chat Rooms
-----------
+Broadcast from node to services
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Raiden nodes will push some updates about their state to the Raiden services:
+they can broadcast a :term:`MonitoringRequest` to the Monitoring Services ("MS") when they go
+offline. A Monitoring Service will then submit their balance proof on their behalf.
+A node can also publish a :term:`PFSCapacityUpdate` and :term:`PFSFeeUpdate` to the PFSes. With this information the PFSes can
+compute efficient routes throughout the network and provide these routes to requesting nodes.
 
-Peer discovery room
-'''''''''''''''''''
-One per network. Participants can discover peers willing to open more channels. It may be implemented in the future as one presence/peer discovery room per token network, but it'd complicate the room-ownership/creation/server problem (rooms need to belong to a server. Whose server? Who created it? Who has admin rights for it?).
+On the transport level, this one-way flow of information is conceptually a broadcast from the node to all services of a specific kind.
+Internally, the broadcast is implemented as several individual to-device messages to all services of a specific
+kind. Here, the ``userId`` of each service registered on-chain is constructed deterministically from the node, so that:
 
-Monitoring Service Updates Room
-'''''''''''''''''''''''''''''''
-Raiden nodes can submit a :term:`MonitoringRequest` to the Monitoring Service room when they go
-offline. The Monitoring Service will submit their balance proof on their behalf.
+::
+
+        "@<address registered on-chain>:<home server of sender's raiden node>"
+
+
+Also, the ``deviceId`` has to be set to ``PATH_FINDING`` for messages to the PFSes and to ``MONITORING`` for messages 
+to the MSes.
+
+The services therefore have to have one Matrix client listening and registered per registered transport server and the
+``deviceId`` of those clients has to be set to ``PATH_FINDING`` or ``MONITORING`` accordingly.
+
+WebRTC messaging
+''''''''''''''''
+To further optimize the communication, exchanging messages in a peer-to-peer
+manner is possible with WebRTC. In that case matrix is only used to initiate the
+WebRTC connection via P2P ``toDevice`` messages, and successive communication between 
+nodes is handled over a webRTC data-channel.
+
+Nodes that support WebRTC messages signal this functionality with the ``webRTC`` capability.
+The requirement is optional, but lack thereof will reduce transfer speeds significantly.
 
```

Comment:
Maybe we should add our custom WebRTC signaling protocol here? Maybe something like
```suggestion

To establish a WebRTC data-channel with another peer, the clients are expected to:
1. Get peer's presence/address-metadata information, either from PFS's specific endpoint or from a passing-through `LockedTransfer`, and has `webRTC` capability
2. Signaling messages are sent to peer's `userId` using `toDevice` matrix messages, with `type` as `m.room.message` (even though it's not in a room) and `content` as `{ "msgtype": "m.notice", "body": <payload> }`, where `payload` is the JSON-encoded string of an object in the format `{ "type": <signal_type>, "call_id": <dataChannel.label>, ...<rest of payload> }`
3. In parallel, start a call of its own, and also listen for calls/offers from peers of interest:
  - On the caller side:
    1. create a `RTCPeerConnection` and a `dataChannel` on it, with whatever label is desired, to uniquely identify this channel upon related messages; an example: `0xCaller_address|0xCallee_address|timestamp`;
    2. start listening for `ICECandidates` on this connection, and send them to peer, with `type="candidates"` and a `candidates` payload member containing an array with the gathered candidates;
    3. create an `offer`, set it as `local description` on connectin, and send it to peer, with `type="offer"` and a `sdp` payload member containing the offer string;
    4. wait for an `answer` message from peer, and upon receiving it, set it as `remote description`; `RTCDataChannel` should then become `open`
    5. a timeout may be put, to retry if neither this call nor callee's side managed to get a channel opened
  - On the callee side:
    1. listen for `offer` messages
    2. when receiving one from a peer of interest, create an `RTCPeerConnection` and set `offer` as `remote description`;
    3. start listening for `ICECandidates` on this connection, and send it to peer, the same as on caller's side
    3. create `answer`, set it as `local description` and send it to peer, with `type="answer"` and `sdp` payload member containing the answer string
    4. wait for `dataChannel` to be emitted and to become `open`
    5. listening for offers on callee is permanent, and any new offer coming through, if succeed, may disconnect previous callee or caller channels;
4. Both caller and callee's codepaths can race; the winner of this race (first channel to become `open`) for each peer's (by address) will disconnect the other direction, and this now-open `RTCDataChannel` will be kept and used for this partner's messaging
5. Upon channel error or close, peers may send a `type="hangup"` message, without additional payload members, and then possibly retry the loop above;
6. Clients may retry whenever they want, if the peer is online; any new open RTC channel for each peer disconnects the previous one; usually, it's ok to retry this just a couple of times and give up, as partner seems to be offline or not responding, and assume they'll call when they come back online; additionally, they may trigger the loop again upon certain events, as new raiden `ChannelOpen` is detected or a message needs to be sent;
```

Reply:
Yes, perfect - thank you! I will format and add this as is (content wise) since I don't know too much about the details of our rtc-signalling